### PR TITLE
fix Flagger connection to Prometheus for Skipper ingress 

### PR DESCRIPTION
**Pull request by dhohengassner**
**enable skipper ingress extended metrics**

Skipper observer is using ingress metrics like
skipper_response_duration_seconds_bucket.
To get this metrics from skipper a few arguments
needs to be set:

https://opensource.zalando.com/skipper/operation/operation/#metrics

**🚚 move skipper ingress kustomization into kustomize dir**

This commit moves the skipper kustomization tested in skipper e2e tests.

During startup Flagger throws an warning because Prometheus is not ready
at this time.
http://flagger-prometheus:9090 unreachable running query failed

There is no need to fix this because it will retried with the Canary.
https://github.com/weaveworks/flagger/issues/664

### CODE REVIEW

**Comment by dhohengassner** [comment]
@universam1 worked on that with Sander - please review

We think we also understood now where the kustomize templates should live.

File: kustomize/skipper/prometheus-skipper/namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: flagger-system
+  annotations:
+    linkerd.io/inject: disabled
```
Comment (universam1) [inline kustomize/skipper/prometheus-skipper/namespace.yaml]: Do we want these annotations?

File: kustomize/skipper/prometheus-skipper/namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  labels:
+    istio-injection: disabled
```
Comment (universam1) [inline kustomize/skipper/prometheus-skipper/namespace.yaml]: Those labels sound unrelated to skipper?

**Review by universam1 — CHANGES_REQUESTED** [review]

File: kustomize/skipper/flagger/patch.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  template:
+    spec:
+      containers:
+        - name: flagger
+          args:
+            - -log-level=info
+            - -mesh-provider=kubernetes
+            - -metrics-server=http://flagger-prometheus:9090
+            - -slack-user=flagger
```
Comment (universam1) [inline kustomize/skipper/flagger/patch.yaml]: we should probably remove these lines when not used

File: kustomize/skipper/prometheus-skipper/namespace.yaml (hunk 2 above)
Comment (dhohengassner) [reply kustomize/skipper/prometheus-skipper/namespace.yaml]: thanks - removed

File: kustomize/skipper/flagger/patch.yaml (hunk 3 above)
Comment (dhohengassner) [reply kustomize/skipper/flagger/patch.yaml]: thanks - removed

**Comment by dhohengassner** [comment]
@universam1 adjusted also based on https://github.com/weaveworks/flagger/issues/664

please review

**Comment by universam1** [comment]
looks good 
LGTM

**Pull request merged**